terraform-modules/stackdriver/k8s-cluster-monitoring/README.md

```diff
@@ -0,0 +1,58 @@
+
+This set of alerting policies is intended to be consistent across all clusters managed by ap-devops. All configuration regarding which metrics the alert policies monitor and the thresholds needed to trigger alerts are managed by locals to ensure they are not overidden accidentally. If conditions or thresholds for these policies need to be added or updated this should be handled by ap-devops.
+
+THe only parameter users of this module need to provide is the id of the gcp project the alerts will be created in. An example usage of the module is provided below. 
+
+```
+module "k8s-cluster-alerts" {
+  source = "https://github.com/broadinstitute/terraform-shared/path/to/this/module"
+  providers = {
+    google.target = google.target
+  }
+
+  project = "YOUR_GCP_PROJECT_ID"
+}
+```
+
+## Policy Documentation
+
+### K8s-cluster-health
+This policy will trigger if the number of pods running on a cluster is approaching the maximum number of pods the cluster can support. This can indicate problems where the scheduler may start evicting crucial pods in order to make more room. This alert is intended and as final line of defense and issues should be caught by lower level alerts before propagating to this level. If this ratio is above 95% for more than 5 minutes an alert will be triggered. This policy utilizes prometheus metrics
+
+### K8s-deployment-health
+This policy monitors the ratio of pods a deployment is currently running vs the number of pods requested in the deployment spec. If these numbers do not match and an update is in progress then this is indicative of an issue and an alert will be fired if this ratio is not equal to 1 for greater than 5 minutes. The alert will provide details about which cluster and namespace are associated with the deployment that triggered the alert. This policy utilizes prometheus metrics
+
+### K8s-node-health
+This policy monitors a variety of different indicators of node health using passing/failing states. These indicators are: 
+- Corrupt Docker Overlay
+- Disk Pressure
+- Memory Pressure
+- Frequent Container Restarts
+- Frequent Docker Restarts
+- Frequent Kubelet Restarts
+- Kernel Deadlock
+- Network Unavailable
+- PID Pressure
+
+If any of these are in a failing state for more than 5 minutes an alert will fire. This policy utilizes prometheus metrics
+
+### k8s-node-resource-utilization
+This policy monitors the cpu and memory utilization ratios of each node in a cluster. This alert policy will trigger if either resource utilization is above a set threshold percentage for greater than 5 minutes. Not reliant on prometheus
+
+### K8s-pod-health
+This policy monitors for any pods that are not able to be run by the scheduler for an extended period of time. This alert fires if a pod is in failed to schedule state for greater than 5 minutes. The alert will specify which cluster and namespace are associated with the pod that triggered the alert.
+
+### K8s-pod-resource utilization 
```

Comment:
Make sure this can be turned off for certain pods.  Sometimes services with memory leaks need to be deployed, and then they use the kubernetes OOM kill process to flush it with a pod restart as part of a normal operational process.

Reply:
Hmm I know you can definitely apply filters, by k8s resources such as pod name or deployment name. However these filters are baked into the metric strings that are found in the locals section of each file. Would have to do some thinking on how to make this toggleable for specific pods. Perhaps an optional variable that adds filters to the end of the string if specified

Reply:
THe easiest way might be to have any service that does this tag its pods with a specific label, then we could filter for that label